app/org/sagebionetworks/bridge/services/AccountWorkflowService.java

```
@@ -225,17 +227,45 @@ public void sendEmailVerificationToken(Study study, String userId, String recipi
+        checkArgument(isNotBlank(userId));
+        
+        if (phone == null) {
+            return;
+        }
+        if (isRequestThrottled(ThrottleRequestType.VERIFY_PHONE, userId)) {
+            // Too many requests. Throttle.
+            return;
+        }
+        String sptoken = getNextToken();
+        String formattedSpToken = sptoken.substring(0,3) + "-" + sptoken.substring(3,6);
+        
+        SmsMessageProvider provider = new SmsMessageProvider.Builder()
+                .withStudy(study)
+                .withToken("sptoken", formattedSpToken)
+                .withSmsTemplate(study.getVerifyPhoneSmsTemplate())
+                .withExpirationPeriod(PHONE_VERIFICATION_EXPIRATION_PERIOD, VERIFY_OR_RESET_EXPIRE_IN_SECONDS)
+                .withPhone(phone).build();
+        notificationsService.sendSMSMessage(provider);
+    }
+        
     /**
-     * Send another email verification token. This creates and sends a new verification token 
-     * starting with the user's email address.
+     * Send another verification token via email or phone. This creates and sends a new verification token 
+     * using the specified channel (an email or SMS message).
      */
-    public void resendEmailVerificationToken(AccountId accountId) {
+    public void resendVerificationToken(ChannelType type, AccountId accountId) {
+        checkNotNull(type);
         checkNotNull(accountId);
         
         Study study = studyService.getStudy(accountId.getStudyId());
         Account account = accountDao.getAccount(accountId);
         if (account != null) {
-            sendEmailVerificationToken(study, account.getId(), account.getEmail());
+            if (type == ChannelType.EMAIL) {
+                sendEmailVerificationToken(study, account.getId(), account.getEmail());
+            } else if (type == ChannelType.PHONE) {
+                sendPhoneVerificationToken(study, account.getId(), account.getPhone());
+            }
```

Should we have an else that throws a "impossible code path" exception? This way, if we ever add another channel and don't update this code, we'll get an exception instead of a silent failure.

---

Seems good, I'll add everywhere we branch like this.